design/node_gateway.md

```diff
@@ -0,0 +1,79 @@
+# Node gateway design
```

Comment:
wouldn't a better name be: "Wallet extension"?

wdyt @CaisManai ?

Reply:
Changed.